src/Rollbar.php

```diff
@@ -249,13 +249,8 @@ public static function logUncaught(string|Level $level, Throwable $toLog, array
         if (is_null(self::$logger)) {
             return self::getNotInitializedResponse();
         }
-        $toLog->isUncaught = true;
-        try {
-            $result = self::$logger->report($level, $toLog, $context);
-        } finally {
-            unset($toLog->isUncaught);
-        }
-        return $result;
+        $wrapper = new ExceptionWrapper($toLog, isUncaught: true);
+        return self::$logger->report($level, $wrapper, $context);
```

Comment:
Suggestion: 

Why not add another optional parameter to `report` that is `bool $uncaught = false`? I find it better because is only that method the one needing that type of information, besides I find it confusing that we pass a `string | Stringeable` type to `report` but it internally coerces it into `mixed` when passing it to `isUncaughtLogData`.

If we do this then wrapper types for the exceptions and `isUncaughtLogData` would no longer be needed. The problem is `isUncaughtLogData` is public 😅 but I'm failing to see why is it useful hence my desire to remove it altogether.

Reply:
That is a really good question.

I was trying to keep `report()` as close to `log()` as possible. I had not considered adding an optional argument, but that would definitely work.

`ExceptionHandler` currently calls the `RollbarLogger::log()` method, but that could be updated to use `report()`, if we went with the optional argument.

Overall, I think that using the optional argument solution would probably reduce complexity. I like it! I'll work on that.